server/block/sugar_cane.go

```diff
@@ -0,0 +1,111 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+)
+
+// SugarCane is a plant block that generates naturally near water.
+type SugarCane struct {
+	empty
+	transparent
+
+	// Age is the growth state of suger cane. Values range from 0 to 15.
+	Age int
+}
+
+// UseOnBlock handles making sure a neighbouring blocks contains water.
+func (c SugarCane) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.World, user item.User, ctx *item.UseContext) (used bool) {
+	pos, _, used = firstReplaceable(w, pos, face, c)
+	if !used {
+		return false
+	}
+	if !c.CanGrowHere(pos, w) {
+		return false
+	}
+
+	place(w, pos, c, user, ctx)
+	return placed(ctx)
+}
+
+// NeighbourUpdateTick ...
+func (c SugarCane) NeighbourUpdateTick(pos, _ cube.Pos, w *world.World) {
+	if !c.CanGrowHere(pos, w) {
+		w.SetBlock(pos, nil, nil)
+		w.AddParticle(pos.Vec3Centre(), particle.BlockBreak{Block: c})
+	}
+}
```

Comment:
You also need to drop the item when it breaks

Reply:
drops item now.